playground/worker/vite.config-sourcemap.js

```diff
@@ -1,12 +1,13 @@
-const vite = require('vite')
-const workerPluginTestPlugin = require('./worker-plugin-test-plugin')
+// @ts-check
+import vite from 'vite'
+import workerPluginTestPlugin from './worker-plugin-test-plugin'
 
-module.exports = vite.defineConfig((sourcemap) => {
-  sourcemap = process.env.WORKER_MODE || sourcemap
-  if (sourcemap === 'sourcemap') {
-    sourcemap = true
-  }
-  return {
+/** @param {boolean | 'inline' | 'hidden'} sourcemap */
+export default (sourcemap) => {
+  sourcemap =
+    /** @type {'inline' | 'hidden'} */ (process.env.WORKER_MODE) || sourcemap
```

Comment:
I think we need to keep the `if (sourcemap === 'sourcemap')` condition here. As `WORKER_MODe` could be `'sourcemap'`, passed for the CLI.